Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -317,58 +317,91 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::index_of)
     return index.has_value() ? Value(*index) : Value(-1);
 }
 
-static ThrowCompletionOr<String> resolve_best_locale(GlobalObject& global_object, Value locales)
-{
-    // For details on these steps, see https://tc39.es/ecma402/#sup-string.prototype.tolocalelowercase
+enum class TargetCase {
+    Lower,
+    Upper,
+};
 
-    // 3. Let requestedLocales be ? CanonicalizeLocaleList(locales).
+// 19.1.2.1 TransformCase ( S, locales, targetCase ), https://tc39.es/ecma402/#sec-transform-case
+static ThrowCompletionOr<String> transform_case(GlobalObject& global_object, String string, Value locales, TargetCase target_case)
```

Comment:
I don't think we need to take the string by value here? Perhaps even a `StringView` is fine, that's what `to_unicode_{lower,upper}case_full()` uses anyway.

Reply:
Sure, I made it a string just to `move` the value in at the callsites below, but yeah it can be just a `StringView`